include/ghex/unstructured/user_concepts.hpp

```diff
@@ -351,6 +351,9 @@ class data_descriptor<ghex::cpu, DomainId, Idx, T>
         assert(!(outer_stride) || (outer_stride >= (levels_first ? m_levels : m_domain_size)));
     }
 
+    /** @brief empty constructor*/
+    data_descriptor(){}
```

Comment:
why are the default constructors needed, i.e. what changed?

Reply:
In the DaCe generated code, I want to create a data_descriptor global variable, because if the field descriptor goes out of scope the asynchronous communication fails. Here is a generated code snippet:
```
ghex::unstructured::data_descriptor<ghex::cpu, int, int, double> field_desc_0_2_5023416384;

inline void _halo_exchange__0_23_0(icon4py_model_atmosphere_diffusion_diffusion_fuse_state_t *__state, int&  __return, double* __restrict__ field_0) {

    {

        {
            double* IN_field_0 = &field_0[0];
            double* OUT_field_0 = field_0;
            int __out;

            ///////////////////

            ghex::context* m = reinterpret_cast<ghex::context*>(__context_ptr_5023416384);

            ghex::pattern_container<ghex::unstructured::detail::grid<unsigned long>, int>* pattern = reinterpret_cast<ghex::pattern_container<ghex::unstructured::detail::grid<unsigned long>, int>*>(__pattern_EdgeDim_ptr_5023416384);
            ghex::unstructured::domain_descriptor<int, int>* domain_descriptor = reinterpret_cast<ghex::unstructured::domain_descriptor<int, int>*>(__domain_descriptor_EdgeDim_ptr_5023416384);
            ghex::communication_object<ghex::unstructured::detail::grid<unsigned long>, int>* communication_object = reinterpret_cast<ghex::communication_object<ghex::unstructured::detail::grid<unsigned long>, int>*>(__comm_ptr_5023416384);


            field_desc_0_2_5023416384 = ghex::unstructured::data_descriptor<ghex::cpu, int, int, double>{*domain_descriptor, IN_field_0, 65, false, 16072};


            h_5023416384 = communication_object->exchange((*pattern)(field_desc_0_2_5023416384));


            __out = 1234; // Dummy return;

            ///////////////////

            __return = __out;
        }

    }
    
}
```
In this way, I keep alive the `field_desc_0_2_5023416384` and I can perform the `h_5023416384.wait();` in another block.

Is there any other way to do thsi without introducing the empty constructor?

Reply:
well, you can always dynamically allocate instead (i.e. use a `unique_ptr`) - would that be acceptable?